src/main/java/io/github/dgroup/arch4u/pmd/UseExistingConstant.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2019-2022 Yurii Dubinka
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom
+ * the Software is  furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
+ * OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package io.github.dgroup.arch4u.pmd;
+
+import java.util.function.Supplier;
+import java.util.regex.Pattern;
+import net.sourceforge.pmd.lang.java.ast.ASTLiteral;
+import net.sourceforge.pmd.lang.java.rule.AbstractJavaRule;
+import net.sourceforge.pmd.lang.java.rule.regex.RegexHelper;
+import net.sourceforge.pmd.properties.PropertyDescriptor;
+import net.sourceforge.pmd.properties.PropertyFactory;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Rule to avoid creating string constants for {@code MediaType} values.
+ * Example: {@code "application/json'}
+ * Use existing classes.
+ *
+ * @since 0.4.0
+ * @see
+ * <a href="https://github.com/dgroup/arch4u-pmd/discussions/43">https://github.com/dgroup/arch4u-pmd/discussions/43</a>
+ */
+@SuppressWarnings("PMD.StaticAccessToStaticFields")
+public final class UseExistingConstant extends AbstractJavaRule {
+
+    /**
+     * Property descriptor with regexp of prohibited string.
+     */
+    private static final PropertyDescriptor<String> REGEX_PROPERTY =
+        PropertyFactory.stringProperty("regexPattern")
+            .desc("Regular expression of prohibited string")
+            .defaultValue(StringUtils.EMPTY)
+            .build();
+
+    /**
+     * Pattern supplier.
+     */
+    private final Supplier<Pattern> pattern;
+
+    /**
+     * Constructor for defining property descriptor.
+     */
+    @SuppressWarnings("PMD.ConstructorOnlyInitializesOrCallOtherConstructors")
+    public UseExistingConstant() {
+        this.definePropertyDescriptor(REGEX_PROPERTY);
+        this.addRuleChainVisit(ASTLiteral.class);
+        this.pattern = () -> Pattern.compile(this.getProperty(REGEX_PROPERTY));
```

Comment:
I am afraid that a pattern will be compiled each time we invoke the supplier (read [more](https://stackoverflow.com/questions/35331327/does-java-8-have-cached-support-for-suppliers/61181887)).
Its better to wrap it by Sticky scalar from cactoos:
```java
this.pattern = new Sticky<>(() -> Pattern.compile(....));
```

Reply:
unfortunately, you're right... it invokes `Pattern#compile` every time `pattern.get` is invoked